### PR TITLE
@ #24 | warning for duplicated hostname

### PR DESCRIPTION
**Pull request by datphan**
connect #24

### CODE REVIEW

File: lib/teracy-dev-essential/config/host_manager.rb [hunk 1]
```diff
@@ -37,10 +43,43 @@ def configure_node(settings, config)
...
+
+        check_conflict_hostname @plugin_aliases
       end
 
       private
 
+      def check_conflict_hostname aliases
+        @logger.debug('check conflict hostname')
+
+        etc_hosts = `cat /etc/hosts`
```
Comment (hoatle) [inline lib/teracy-dev-essential/config/host_manager.rb]: this path is different between OSes, especially Windows, make sure this is portable.

File: lib/teracy-dev-essential/config/host_manager.rb [hunk 2]
```diff
@@ -9,6 +9,12 @@ class HostManager < TeracyDev::Config::Configurator
       def configure_common(settings, config)
         @plugins = settings['vagrant']['plugins'] ||= []
 
+        plugin = @plugins.find { |p| p['name'] == PLUGIN_NAME }
+
+        @plugin_aliases = []
+
+        @plugin_aliases = plugin['options']['aliases'] if plugin
```
Comment (hoatle) [inline lib/teracy-dev-essential/config/host_manager.rb]: this config could come from common config or each node config, so we need to check on both.

Comment (hoatle) [reply lib/teracy-dev-essential/config/host_manager.rb]: see this config for example: https://github.com/teracyhq-incubator/teracy-dev-entry-k8s/blob/develop/config_default.yaml#L75

Comment (datphan) [reply lib/teracy-dev-essential/config/host_manager.rb]: @hoatle I've updated, pls check again

**Comment by hoatle** [comment]
@phuonglm pls help reviewing this PR

File: lib/teracy-dev-essential/config/host_manager.rb [hunk 3]
```diff
@@ -36,11 +49,70 @@ def configure_node(settings, config)
...
+          return false
+        end
+
+        @logger.debug("etc_hosts: #{etc_hosts}")
+
+        conflict_list = []
+        
+        hex = "[0-9a-fA-F]+"
+        sign = "\.?\:?"
+        ip = "#{hex}#{sign}#{hex}#{sign}#{hex}#{sign}#{hex}#{sign}#{hex}#{sign}#{hex}#{sign}#{hex}#{sign}#{hex}"
```
Comment (phuonglm) [inline lib/teracy-dev-essential/config/host_manager.rb]: Could you give example of which kind of ip you looking for with this regex? IPV4 IPV6.
For me it look like `10.0.0.13` doesn't match this regex

Comment (datphan) [reply lib/teracy-dev-essential/config/host_manager.rb]: thanks for the check, I've updated it, pls check again

Sample hosts file I'm looking for:

```


192.168.1.86	dev.iorad.local
192.168.1.86	node-01.local


127.0.0.1       dev.iorad.local

2405:4800:528f:ed7:fb:d787:6821:a68a dev.iorad.local

2405:4800:528f:ed7:c5c1:1dae:a7a:d94c node-01.local
```

**Pull request merged**